### ISSUE TITLE
CQL spatial filters must include the SRID of the GeometryLiteral's SRS

### ISSUE THREAD

**Issue by giohappy** [geosolutions-it/MapStore2#4277]
### Description
Geoserver applies CQL spatial filters according to the target layer's SRS. When MapStore sends WMS requests with a CQL spatial filter the SRS of the filter's geom coordinates must be passed along so that Geoserver applies the required transformations.

For instance:

(INTERSECTS(the_geom, SRID=3857;Polygon((94

*Steps to reproduce*

Run the following request, where target layer is in EPSG:32632
```
https://dev.geonode.geo-solutions.it/geoserver/ows?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2Fpng8&TRANSPARENT=true&LAYERS=geonode%3AReg01012019_g_WGS84&STYLES=&SRS=EPSG%3A3857&CRS=EPSG%3A3857&TILED=true&access_token=VCCYMMlDqEKvH7uOjotJ6eK6m650CL&WIDTH=256&HEIGHT=256&BBOX=939258.2035682462%2C4696291.017841227%2C1017529.7205322667%2C4774562.534805248&CQL_FILTER=(INTERSECTS(the_geom%2C%20Polygon((948889.2691321779%204718916.37821364%2C%20948889.2691321779%204694456.529162384%2C%201004841.1738369269%204694456.529162384%2C%201004841.1738369269%204718916.37821364%2C%20948889.2691321779%204718916.37821364))))
```

*Expected Result*
A red feature should be visible:

![image](https://user-images.githubusercontent.com/571129/66033927-88da0780-e508-11e9-87fe-dceb021c622d.png)

*Current Result*
Empty image

![image](https://user-images.githubusercontent.com/571129/66034107-dc4c5580-e508-11e9-81a7-e498b96ab2f6.png)

### Other useful information:
The same query as before, with the SRID prepended, works:

```
https://dev.geonode.geo-solutions.it/geoserver/ows?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2Fpng8&TRANSPARENT=true&LAYERS=geonode%3AReg01012019_g_WGS84&STYLES=&SRS=EPSG%3A3857&CRS=EPSG%3A3857&TILED=true&access_token=VCCYMMlDqEKvH7uOjotJ6eK6m650CL&WIDTH=256&HEIGHT=256&BBOX=939258.2035682462%2C4696291.017841227%2C1017529.7205322667%2C4774562.534805248&CQL_FILTER=(INTERSECTS(the_geom%2C%20SRID%3D3857%3BPolygon((948889.2691321779%204718916.37821364%2C%20948889.2691321779%204694456.529162384%2C%201004841.1738369269%204694456.529162384%2C%201004841.1738369269%204718916.37821364%2C%20948889.2691321779%204718916.37821364))))
```

**Comment by kappu72** [comment]
Does It works with any geoserver version or is It a recent improvement? 
If latter case, what's the expected behaviour. (just considering last geoserver version or try to manage the different cases).
As far as I remember in Ms2 a cql spatial filter is reprojected in layers orig-srs, if this info it's available.

**Comment by kappu72** [comment]
The issue comes from [django-mapstore-adapter ](https://github.com/GeoNode/django-mapstore-adapter/blob/master/mapstore2_adapter/plugins/geonode.py#L269). The adapter has to return the correct (saved) layer nativeCrs. Here when missing it returns the map crs!
To fix:
if 'nativeCrs' in layer 
   overlay['nativeCrs'] = layer['nativeCrs']

**Comment by giohappy** [comment]
@tdipisa this issue can be turned to a feature rather then a bug.

**Comment by tdipisa** [comment]
closed in favour of #4306
